crypto/fipsmodule/evp/evp.c

```diff
@@ -263,6 +263,35 @@ void evp_pkey_set_method(EVP_PKEY *pkey, const EVP_PKEY_ASN1_METHOD *method) {
   pkey->type = pkey->ameth->pkey_id;
 }
 
+static int pkey_set_type(EVP_PKEY *pkey, int type, const char *str, int len) {
+  if (pkey && pkey->pkey.ptr) {
+    // This isn't strictly necessary, but historically |EVP_PKEY_set_type| would
+    // clear |pkey| even if |evp_pkey_asn1_find| failed, so we preserve that
+    // behavior.
+    free_it(pkey);
+  }
+
+  const EVP_PKEY_ASN1_METHOD *ameth = NULL;
+
+  if (str != NULL) {
+    ameth = EVP_PKEY_asn1_find_str(NULL, str, len);
+  } else {
+    ameth = evp_pkey_asn1_find(type);
+  }
+
+  if (ameth == NULL) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_UNSUPPORTED_ALGORITHM);
+    ERR_add_error_dataf("algorithm %d", type);
```

Comment:
`ERR_add_error_dataf` is used throughout our libssl, but seldom so in the `fipsmodule`

https://github.com/search?q=repo%3Aaws%2Faws-lc%20ERR_add_error_dataf&type=code

question to other reviewers -- is this a pattern we should lean in to?

Reply:
`ERR_add_error_dataf` is used in this file for the same purpose on [line 458](https://github.com/aws/aws-lc/blob/9921cd9694221c1107210f9c4f229b3d2f5acd9c/crypto/fipsmodule/evp/evp.c#L458). 

We're not consistent with using it though, so I doubt consumers are relying on this error message to be provided. However, I'm also not aware of any reason for us to avoid using `ERR_add_error_dataf` in the fipsmodule. I'm fine with keeping it.